x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/InferModelAction.java

```diff
@@ -23,44 +29,87 @@
 import java.util.List;
 import java.util.Map;
 import java.util.Objects;
+import java.util.stream.Collectors;
 
-public class InternalInferModelAction extends ActionType<InternalInferModelAction.Response> {
-
-    public static final InternalInferModelAction INSTANCE = new InternalInferModelAction();
+public class InferModelAction extends ActionType<InferModelAction.Response> {
     public static final String NAME = "cluster:internal/xpack/ml/inference/infer";
+    public static final String EXTERNAL_NAME = "cluster:monitor/xpack/ml/inference/infer";
```

Comment:
We needed an external name for interaction at the REST layer and since inference needs to work during rolling upgrade, we cannot get rid of the old action name.

So, both point to effectively the same transport class.